### PR TITLE
ci: Enable LeakSanitizer in bindings-rust-asan

### PR DESCRIPTION
**Pull request by axic**
Part of #614.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -691,7 +691,7 @@ jobs:
           command: |
             export RUSTFLAGS="-Z sanitizer=address -C opt-level=0"
             export RUSTDOCFLAGS=$RUSTFLAGS
-            cargo +nightly test --target x86_64-unknown-linux-gnu
+            ASAN_OPTIONS=detect_leaks=1 cargo +nightly test --target x86_64-unknown-linux-gnu
```
Comment (axic) [inline circle.yml:694]: Tested this with introducing a leak: https://app.circleci.com/pipelines/github/wasmx/fizzy/6898/workflows/fa937f8b-c3b3-4291-bbb3-2550ab9d2e94/jobs/62254

Comment (gumb0) [inline circle.yml:694]: why not use `export` as in above lines?

Comment (axic) [reply circle.yml]: Moved this around, still works as expected: https://app.circleci.com/pipelines/github/wasmx/fizzy/6906/workflows/e190b481-3c7e-4cf6-a7cd-15f26b69eb7b/jobs/62392

**Comment by axic** [comment]
As feared the leak sanitizer is on by default now: https://app.circleci.com/pipelines/github/wasmx/fizzy/6907/workflows/0618375e-8c94-495c-8477-8f7ed74f9521/jobs/62402

**Comment by axic** [comment]
It seems to be on-by-default, also suggested by https://clang.llvm.org/docs/LeakSanitizer.html.

**Pull request closed without merging**